lms/static/scripts/frontend_apps/components/InitialLoadingTracker.tsx

```diff
@@ -0,0 +1,76 @@
+import { Spinner } from '@hypothesis/frontend-shared';
+import classnames from 'classnames';
+import type { ComponentChildren } from 'preact';
+import { useCallback, useMemo, useRef, useState } from 'preact/hooks';
+
+import { InitialLoadingContext } from '../utils/initial-loading-context';
+
+export type InitialLoadingProps = {
+  children: ComponentChildren;
+  classes?: string | string[];
+};
+
+export default function InitialLoadingTracker({
+  children,
+  classes,
+}: InitialLoadingProps) {
+  const [initialLoadingInProgress, setInitialLoadingInProgress] =
+    useState(true);
+  const loadingInstances = useRef<Record<string, boolean>>({});
```

Comment:
Could this just use a `Set`, to track the remaining pending loads?

Reply:
A set with the `id`s? The only thing would be being able to differentiate between "it's empty because no individual loads started yet" vs "all loads finished". The record allows to differentiate these two.

Reply:
Actually, differentiating that is not needed. It was in a previous iteration where `initialLoadInProgress` was derived state, but now it's not anymore.

I'll try with the Set.

Reply:
Yep, a set does the job, assuming `startLoading` will always be invoked before `finishLoading` for a specific key.